src/include/sof/math/trig.h

```diff
@@ -11,11 +11,35 @@
 #define __SOF_MATH_TRIG_H__
 
 #include <stdint.h>
+#include <sof/bit.h>
+
+#ifndef UNIT_CORDIC_TEST
+#define CONFIG_CORDIC_TRIGONOMETRY_FIXED
+#else
+#define CONFIG_GENERIC_TRIGONOMETRY_FIXED
+#endif
+
 
 #define PI_DIV2_Q4_28 421657428
 #define PI_Q4_28      843314857
 #define PI_MUL2_Q4_28     1686629713
+#define CORDIC_31B_TABLE_SIZE		31
+#define CORDIC_SIN_COS_15B_TABLE_SIZE	15
+
+typedef enum {
+	EN_32B_CORDIC_SINE,
+	EN_32B_CORDIC_COSINE,
+	EN_16B_CORDIC_SINE,
+	EN_16B_CORDIC_COSINE,
+} cordic_cfg;
 
-int32_t sin_fixed(int32_t th_rad_fxp); /* Input is Q4.28, output is Q1.31 */
+void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t *b_yn, int32_t *xn,
+		    int32_t *th_cdc_fxp);
+/* Input is Q4.28, output is Q1.31 */
+int32_t sin_fixed_32b(int32_t th_rad_fxp,  cordic_cfg  type);
+int32_t cos_fixed_32b(int32_t th_rad_fxp,  cordic_cfg  type);
+/* Input is Q4.28, output is Q1.15 */
+int16_t sin_fixed_16b(int32_t th_rad_fxp, cordic_cfg  type);
+int16_t cos_fixed_16b(int32_t th_rad_fxp, cordic_cfg  type);
```

Comment:
Please remove argument "type", it's not needed. You can set it inside these functions for cordic_sin_cos() function call.